csharp/src/Google.Protobuf/CodedInputReader.cs

```diff
@@ -86,7 +86,7 @@ internal CodedInputReader(ReadOnlySequence<byte> input, int recursionLimit)
             this.lastTag = 0;
             this.recursionDepth = 0;
             this.recursionLimit = recursionLimit;
-            this.currentLimit = long.MaxValue;
+            this.currentLimit = (int)this.reader.Length;
```

Comment:
Why not slice the input upfront?

Reply:
Slice what up front?

Setting the current limit here lets us use this value as "all the remaining content". It avoids an additional check to long.MaxValue when testing the current limit.